unittests/UnitTestFramework.xml

```diff
@@ -171,12 +188,66 @@ begin
 				errorline = p.linenumber,
 				errorexec =	p.Name);
 		endif;
+	elseif expectedString is not null then
+		if equalStrings(s1=expectedString, s2=actualString)=FALSE then
+			if CurObject.TraceStringsOnFailure=TRUE then
+				CurMethod.Trace(text = HC_NEWLINE+'=== Expected string: ===');
+				CurMethod.Trace(string = expectedString);
+				CurMethod.Trace(text = HC_NEWLINE+'=== Actual string (not equal): ===');
+				CurMethod.Trace(string = actualString);
+			endif;
+			CurObject.throwAssertionFailedError(errortext=errortext+HC_NEWLINE+HC_TAB+
+				'assertEquals: actualString does not match expectedString',
+				errorline = p.linenumber,
+				errorexec =	p.Name);
+		endif;
+	elseif expectedJSON is not null then
+		if equalJSON(json1=expectedJSON, json2=actualJSON)=FALSE then
+			if CurObject.TraceStringsOnFailure=TRUE then
+				CurMethod.Trace(text = HC_NEWLINE+'=== Expected JSON: ===');
+				CurMethod.Trace(string = expectedJSON);
+				CurMethod.Trace(text = HC_NEWLINE+'=== Actual JSON (not equal): ===');
+				CurMethod.Trace(string = actualJSON);
+			endif;
+			CurObject.throwAssertionFailedError(errortext=errortext+HC_NEWLINE+HC_TAB+
+				'assertEquals: actualJSON does not match expectedJSON',
+				errorline = p.linenumber,
+				errorexec =	p.Name);
+		endif;
+	elseif expectedJsonValue is not null then
+		if actualJsonValue is null then
+			CurObject.throwAssertionFailedError(errortext=errortext+HC_NEWLINE+HC_TAB+
+				'assertEquals: actualJsonValue is NULL',
+				errorline = p.linenumber,
+				errorexec =	p.Name);
+		endif;
+		jh = CurSession.JsonHandler;
+		j1 = stringobject.Create();
+		j2 = stringobject.Create();
+		jh.Write(value=expectedJsonValue, json=j1, sortobjectmembers=TRUE);
+		jh.Write(value=actualJsonValue, json=j2, sortobjectmembers=TRUE);

```

Comment:
Could you add indented=TRUE

Reply:
See my comment for previous line.

Reply:
This is a fair statement.  I was thinking how it would visually look not about the real purpose which is comparison.  I will approve it as it is.